game-core/src/main/java/games/strategy/engine/framework/startup/launcher/LocalLauncher.java

```diff
@@ -118,4 +123,33 @@ public static LocalLauncher create(
             null);
     return new LocalLauncher(gameSelectorModel, new PlainRandomSource(), pl, parent, launchAction);
   }
+
+  public static LocalLauncher create(
+      final List<Entry<String, String>> roleMapping,
+      final Predicate<String> isDisabled,
+      final GameData gameData) {
+    final Predicate<Entry<String, String>> enabledPredicate =
+        entry -> !isDisabled.test(entry.getValue());
+    final PlayerListing playerListing =
+        new PlayerListing(
```

Comment:
Is it worthwhile to create constructor over-loads to avoid the `null` args? We might be able to put more of the construction work in `PlayerListing` constructor as well to further simplify here.

Reply:
I don't know if you noticed, but I changed the way how users disable nations in the new UI:
Instead of unchecking a checkbox to disable a player, there is a "Disable" option when choosing who controls the specific nation, which feels more natural IMO.
However `PlayerListing` is used to this checkbox based scheme so I have to convert the data representation so it matches the "old"/current scheme.

So I could add an overload constructor, but I think this would ultimately lead down a rabbit hole of potential improvements for the data scheme in general.

Reply:
I did not quite notice that, there is quite a bit going on in this update and I'm afraid to say time is too limited to spend enough time to really properly review everything without further pointers/guidance to call important elements out.

An overload of `PlayerListing` is pretty limited, why would that lead to a rabbit hole? Just add only the overload, then the class here in question does not have to pass unused/null values.